### PR TITLE
Add widget types

### PR DESCRIPTION
**Pull request by OlivierZal**

### CODE REVIEW

**Comment by OlivierZal** [comment]
@jeroenwienk, can you please review?

**Comment by OlivierZal** [comment]
@jeroenwienk, can you please review?

File: lib/HomeyWidget.d.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+     * @param {string} url
+     */
+    popup(url: string): Promise<void>;
+    /**
+     * The settings view will be hidden until this method has been called.
+     * Use the extra time to make required API calls to prevent flickering on screen.
+     * @param {Object} [args] An object with a height property.
+     * @param {number|string} [args.height] The height of the widget.
+     */
+    ready(args?: { height: number | string }): Promise<void>;
```
Comment (OlivierZal) [inline lib/HomeyWidget.d.ts]: Documentation says `void`, but according to my test it returns a `Promise` (like for the other interfaces).

Comment (jeroenwienk) [reply lib/HomeyWidget.d.ts]: It doesnt return a promise this is the code. So whats going on here :)

`
  ready({ height } = { height: null }) {
    if (this._ready !== true) {
      this._ready = true;
      this._crossFrame.send('ready', { height });
    }
  }
`

Comment (OlivierZal) [reply lib/HomeyWidget.d.ts]: @jeroenwienk,

You raised your case, I changed it.

As per my test, it seems that there was less flickering with `await` than without.

Is it the same for the `ready` of app settings and the other `ready`?

**Review by jeroenwienk — APPROVED** [review]

**Pull request merged**